lib/asyn-thread.c

```diff
@@ -79,6 +79,10 @@
 #include "curl_memory.h"
 #include "memdebug.h"
 
+#ifdef HAVE_SOCKETPAIR
+#define INVALID_SOCKET (-1)
+#endif
+
```

Comment:
Please use `CURL_SOCKET_BAD` instead of `INVALID_SOCKET`. We do that all over the code, which is more portable. Then you don't need this define at all.

Reply:
done.